### DISCUSSION TITLE
Average commits per day seems to be much lower than its actual value

### DISCUSSION THREAD

**Discussion by developomp** [lowlighter/metrics#743] [Q&A] [answered]
I have around 1500 commits pushed in the last 365 days, most of which are concentrated in the last couple of months. However, the average commits per day is 1.44 when it should be at least 4. Why is this the case?

Here's my GitHub profile: https://github.com/developomp

**Comment by matievisthekat** [comment]
whats your configured amount of days to use when calculating the average commits per day?

**Reply by developomp** [reply]
Do you mean the isocalendar type? I don't see any configs for average commit calculation.

**Comment by lowlighter** [comment, marked as answer]
Hi 👋 !
I think it's because the commits gets divided by your account age:
1500/(3*365) ≈ 1.37 ≈ 1.44

Normally the commits count should be your total amount of commits, but it can be limited due to GitHub search api (so it'll fallback to the last year). Because of this, it results in a lower average like you reported.
https://github.com/lowlighter/metrics/blob/fca0c3f49b3cf4ed93a69b2935a1d830aa771372/source/plugins/base/index.mjs#L118-L125

I think the best way to handle this would be to catch whether we have fallbacked or now, and separate into 2 stats (All time, and Windowed to isocalendar duration)

This will be tracked by #753

**Reply by developomp** [reply]
Ah, that makes sense. The calendar beside it made me assume that the metrics is for the past year's data.
